### DISCUSSION TITLE
Missing librechat.yaml docker binding

### DISCUSSION THREAD

**Discussion by michnovka** [danny-avila/LibreChat#3755] [Troubleshooting]
### What happened?

When running `docker compose up`, the api container complains it cannot find /app/librechat.yaml

### Steps to Reproduce

1. clone repo
2. docker compose up

### Relevant log output

```shell
LibreChat         | 2024-08-23 08:36:13 error: Config file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'
```
```


### Screenshots

![image](https://github.com/user-attachments/assets/7a5f8a7b-78a1-498a-94ab-09dbf996b45f)


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by michnovka** [comment]
Adding:

```yaml
services:
  api:
    ....
    volumes:
    ....
      - ./librechat.yaml:/app/librechat.yaml
  
```

fixes the issue

**Reply by tvhahn** [reply]
> Adding:
> 
> ```yaml
> services:
>   api:
>     ....
>     volumes:
>     ....
>       - ./librechat.yaml:/app/librechat.yaml
>   
> ```
> 
> fixes the issue

I had this exact same issue. Your solution worked. Seems to be a bug to me?

Note, I could still log into LibreChat, but then when I tried to use the Azure OpenAI model, it threw up the error `LibreChat         | 2024-08-23 08:36:13 error: Config file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'`

**Comment by danny-avila** [comment]
Not a bug, and still works without it

![image](https://github.com/user-attachments/assets/13882072-2d1b-4a41-b18f-5d89ad6fda73)

**Comment by thisismygitrepo** [comment]
It doesn't work on my side

```ibreChat         | 2024-08-28 01:49:31 error: Config file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'```

**Reply by danny-avila** [reply]
> `ibreChat | 2024-08-28 01:49:31 error: Config file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'`

so the app doesn't start for you?

**Reply by danny-avila** [reply]
it should still work, and you can try setting up the `librechat.yaml` file just in case:
https://www.librechat.ai/docs/quick_start/custom_endpoints

**Reply by alexleducanh** [reply]
I tried the method on the doc. The app start normally but my custom endpoint is not. 
Also i have this error: Config file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'

**Comment by rajvatsal** [comment]
Hey guys @michnovka follow these steps [here](https://www.librechat.ai/docs/configuration/librechat_yaml/setup#docker-setup), this might solve your issue if you haven't done this already.

**Comment by FrancoisMentec** [comment]
I landed here when searching why my `librechat.yaml` was missing from my container so I think it is worth taking the time to answer:

- `librechat.yaml` doesn't exists by default and LibreChat works just fine without it (despite the error message during the build), you only need it if you want to change the default config. For example you may want to add Mistral to your LibreChat.
- To create `librechat.yaml` you can copy `librechat.example.yaml`. You can find instructions about the file here: https://www.librechat.ai/docs/configuration/librechat_yaml
- Now to include `librechat.yaml` in your container you need to create a `docker-compose.override.yml` with this content:
  ```yaml
  services:
    api:
      volumes:
      - type: bind
        source: ./librechat.yaml
        target: /app/librechat.yaml
  ```
  I recommend you copy `docker-compose.override.yml.example` and read the instructions at the top of the file.
- Finally you can rebuild your container and start it as you would normally do.
